modules/@apostrophecms/doc-type/ui/apos/components/AposDocEditor.vue

```diff
@@ -112,27 +85,92 @@ export default {
   },
 
   computed: {
+    moduleOptions() {
+      return window.apos.modules[this.moduleName] || {};
+    },
+    schema() {
+      return this.moduleOptions.schema || [];
+    },
+    groups() {
+      const groupSet = {
+        basics: {
+          label: 'Basics',
+          fields: []
+        }
+      };
+
+      this.schema.forEach(field => {
+        if (field.group && !groupSet[field.group.name]) {
+          groupSet[field.group.name] = {
+            label: field.group.label,
+            fields: [ field.name ]
+          };
+        } else if (field.group) {
+          groupSet[field.group.name].fields.push(field.name);
+        } else {
+          groupSet.basics.fields.push(field.name);
+        }
+      });
+
+      return groupSet;
+    },
+    utilityFields() {
+      if (this.groups.utility && this.groups.utility.fields) {
+        return this.groups.utility.fields;
+      }
+      return [];
+    },
+    tabs() {
+      const tabs = [];
+      for (const key in this.groups) {
+        if (key !== 'utility') {
+          const temp = { ...this.groups[key] };
+          temp.name = key;
+          tabs.push(temp);
+        }
+      };
+      return tabs;
+    },
     modalTitle () {
-      return `Edit ${this.typeLabel}`;
+      return `Edit ${this.moduleOptions.pluralLabel}`;
     },
     currentFields: function() {
-      const fields = [];
       if (this.currentTab) {
-        const tabFields = this.tabs.find((item) => item.name === this.currentTab);
-        tabFields.fields.forEach((field) => {
-          fields.push(this.schema.find(item => field === item.name));
+        const tabFields = this.tabs.find((item) => {
+          return item.name === this.currentTab;
         });
+
+        return tabFields.fields;
+      } else {
+        return [];
       }
-      return fields;
     }
   },
   async mounted() {
-    // TODO: Get data here.
     this.modal.active = true;
+
+    if (this.docId) {
+      // TODO: Get data here.
```

Comment:
This will be done in the "edit a piece" story as well.

Reply:
I am not sure if you are repurposing this for "page settings," but if you are, start thinking ahead to what happens when the user changes the page type. That can change the schema and the tabs on the fly.

Reply:
I moved the schema population into a method to make it reusable in this context and future contexts. I'm not going to do too much business logic for the page settings type case right now since it's not the subject of the current ticket, but I'll make sure to create a ticket for that.